### ISSUE TITLE
This is written in c++

### ISSUE THREAD

**Issue by fsaintjacques** [pkhuong/modok#1]
It seems to me that optimality could be achieved by writing it in a more metal friendly language like C or the lisp flavoured SBCL.

**Comment by postamar** [comment]
Do you know the D language? You should consider porting this to D. It's C++ done right!

**Comment by pkhuong** [comment]
I agree. If anything, this tree shows how one can easily be fooled into using C++ as a worse Python. The next version will be in C.
